packages/subgraph/tests/dao/dao_v1_0_0.test.ts

```diff
@@ -924,71 +963,102 @@ describe('handleExecuted', () => {
+      event.address,
+      event.params.actor,
+      event.params.callId,
+      0,
+      event.transaction.hash,
+      event.transactionLogIndex
+    );
 
     // create proposal
-    let proposal = new TransactionActionsProposal(
-      transactionActionsProposalEntityId
-    );
-    proposal.dao = event.address.toHexString();
-    proposal.createdAt = event.block.timestamp;
-    proposal.endDate = event.block.timestamp;
-    proposal.startDate = event.block.timestamp;
-    proposal.allowFailureMap = BigInt.zero();
-    proposal.creator = event.params.actor;
-    proposal.executionTxHash = event.transaction.hash;
-    proposal.executed = true;
-    proposal.save();
+    let transactionActions = new TransactionActions(transactionActionsEntityId);
+    transactionActions.dao = event.address.toHexString();
+    transactionActions.createdAt = event.block.timestamp;
+    transactionActions.deterministicId = deterministicId;
+    transactionActions.endDate = event.block.timestamp;
+    transactionActions.startDate = event.block.timestamp;
+    transactionActions.allowFailureMap = BigInt.zero();
+    transactionActions.creator = event.params.actor;
+    transactionActions.executionTxHash = event.transaction.hash;
+    transactionActions.executed = true;
+    transactionActions.save();
 
     // create action
-    let action = new Action(actionEntityId);
+    let action = new TransactionAction(actionEntityId);
     action.to = Address.fromString(DAO_TOKEN_ADDRESS);
     action.data = Bytes.fromHexString('0x');
     action.value = BigInt.zero();
     action.dao = event.address.toHexString();
-    action.proposal = proposal.id;
+    action.transactionActions = transactionActions.id;
+    action.deterministicId = deterministicActionId;
     action.save();
 
     // Check that before `handleExecute`, execResults are empty
-    assert.entityCount('Action', 1);
-    assert.entityCount('TransactionActionsProposal', 1);
+    assert.entityCount('TransactionAction', 1);
+    assert.entityCount('TransactionActions', 1);
     assert.assertTrue(action.execResult === null);
-    assert.assertTrue(proposal.failureMap === null);
+    assert.assertTrue(transactionActions.failureMap === null);
 
     handleExecuted(event);
 
     // The action and proposal count should be the same.
-    assert.entityCount('Action', 1);
-    assert.entityCount('TransactionActionsProposal', 1);
+    assert.entityCount('TransactionAction', 1);
+    assert.entityCount('TransactionActions', 1);
 
-    eq('Action', actionEntityId, 'id', actionEntityId);
-    eq('Action', actionEntityId, 'execResult', execResult.toHexString());
+    eq('TransactionAction', actionEntityId, 'id', actionEntityId);
+    eq(
+      'TransactionAction',
+      actionEntityId,
+      'execResult',
+      execResult.toHexString()
+    );
 
     eq(
-      'TransactionActionsProposal',
-      transactionActionsProposalEntityId,
+      'TransactionActions',
```

Comment:
My last comment would be that we lack the very important test. Maybe I am missing it.

Test where 2 transactionActions Entity are created but with the same deterministic id and then checking with assert entity that in the database with deterministic Id, there're 2 records. Same test also has to be in dao_v1_3_0.test.ts

Reply:
I agree that's a very good thing to add